pkg/plugin/status.go

```diff
@@ -0,0 +1,206 @@
+	RecoveryStageDone    = "DONE"
+)
+
+// SnapshotGetResponse is used to unmarshal the response body of SnapshotGetRequest
+type SnapshotGetResponse struct {
+	Snapshots []SnapshotStatus `json:"snapshots"`
+}
+type SnapshotStatus struct {
+	Snapshot string `json:"snapshot"`
+	State    string `json:"state"`
+}
+
+// IndexesRecoveryStatus is used to unmarshal the indexes recovery status from IndicesRecoveryRequest response
+type IndexesRecoveryStatus struct {
+	Shards []ShardRecoveryStatus `json:"shards"`
+}
+type ShardRecoveryStatus struct {
+	Type   string         `json:"type"`
+	Stage  string         `json:"stage"`
+	Source RecoverySource `json:"source"`
+}
+type RecoverySource struct {
+	Snapshot string `json:"snapshot,omitempty"`
+}
+
+func (d *ElasticsearchDDM) Status(params map[string]string) (int32, error) {
+	fmt.Println("\nHandling STATUS request..................")
+	// extract the plugin parameters from the respective MoveEngine
+	pluginParameters, mode, err := extractPluginParameters(d.DmClient, params)
+	if err != nil {
+		return framework.Errored, err
+	}
+
+	snapshotName, found := params[KeySnapshotName]
+	if !found {
+		return framework.Errored, fmt.Errorf("snapshot name not found in the parameters")
+	}
+
+	// if it is active cluster, then send the backup state
+	if mode == EngineModeActive {
+		return retrieveBackupState(d.K8sClient, pluginParameters, snapshotName)
+	} else {
+		// if it is destination cluster, then send restore status
+		return retrieveRestoreState(d.K8sClient, pluginParameters, snapshotName)
+	}
+}
+
+// retrieveSnapshotState hits the Snapshot API of Elasticsearch to retrieve the backup status of a Snapshot
+func retrieveBackupState(k8sClient kubernetes.Interface, params PluginParameters, snapshotName string) (int32, error) {
+	// crate an Elasticsearch client
+	esClient, err := NewElasticsearchClient(k8sClient, params.Elasticsearch)
+	if err != nil {
+		return framework.Errored, err
+	}
+
+	// configure snapshot get request
+	snapshotGetRequest := esapi.SnapshotGetRequest{
+		Repository: params.Repository.Name,
+		Snapshot:   []string{snapshotName},
+		Pretty:     true,
+	}
+
+	// get snapshot status
+	fmt.Println("Requesting for snapshot status.....")
+	resp, err := snapshotGetRequest.Do(context.Background(), esClient)
+	if err != nil {
+		return framework.Errored, err
+	}
+	defer resp.Body.Close()
+
+	fmt.Println("Response: ", resp.String())
+
+	if resp.StatusCode != http.StatusOK {
+		rootCause, err := parseErrorCause(resp.Body)
+		if err != nil {
+			return framework.Errored, fmt.Errorf("failed to retrieve backup state.\n"+
+				"Also, failed to parse the error info. Reason: %s", err.Error())
+		}
+		return framework.Errored, fmt.Errorf("failed to retrieve backup state.\n"+
+			"Error Type: %s, Reason: %s", rootCause.Type, rootCause.Reason)
+	}
+
+	var statusResponse SnapshotGetResponse
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return framework.Errored, err
```

Comment:
Better to wrap this error with a message.

Reply:
done